src/client/components/pages/parts/add-to-collection-modal.js

```diff
@@ -0,0 +1,296 @@
+					</h4>
+					<div className="addToCollectionModal-body">
+						{
+							this.state.collectionsAvailable.map((collection) => ((
+								<div key={collection.id}>
+									<input
+										checked={this.state.selectedCollections.find(selectedId => selectedId === collection.id)}
+										id={collection.id}
+										type="checkbox"
+										onChange={() => this.toggleRow(collection.id)}
+									/>
+									<label className="label-checkbox" htmlFor={collection.id}>
+										{collection.name}
+									</label>
+								</div>
+							)))
+						}
+					</div>
+				</div>
+			);
+		}
+		else {
+			existingCollections = (
+				<div>
+					Oops, looks like you do not yet have any collection of {this.props.entityType}s .
+					Click on the button below to create a new collection
+				</div>
+			);
+		}
+
+		const privacyOptions = ['Private', 'Public'].map((option) => ({
+			name: option
+		}));
+		const collectionForm = (
+			<div>
+				<Col
+					id="collectionForm"
+				>
+					<form
+						className="padding-sides-0 addToCollectionModal-body"
+					>
+						<CustomInput
+							label="Name"
+							ref={(ref) => this.name = ref}
+							type="text"
+						/>
+						<CustomInput
+							label="Description"
+							ref={(ref) => this.description = ref}
+							type="textarea"
+						/>
+						<SelectWrapper
+							base={ReactSelect}
+							idAttribute="name"
+							label="Privacy"
+							labelAttribute="name"
+							options={privacyOptions}
+							placeholder="Select Privacy"
+							ref={(ref) => this.privacy = ref}
+						/>
+					</form>
+				</Col>
+			</div>
+		);
+
+		return (
+			<Modal
+				scrollable
+				show={this.props.show}
+				onHide={this.props.onCloseCallback}
+			>
+				<Modal.Header closeButton>
+					<Modal.Title>
+						Select Collection
+					</Modal.Title>
+				</Modal.Header>
+				<Modal.Body>
+					{
+						this.state.showCollectionForm ? collectionForm : existingCollections
+					}
+					{messageComponent}
+				</Modal.Body>
+				<Modal.Footer>
+					{
+						this.state.showCollectionForm ?
+							<Button
+								bsStyle="primary"
+								onClick={this.handleShowAllCollections}
+							>
+								Select from collections
+							</Button> :
+							<Button
+								bsStyle="warning"
+								onClick={this.handleShowCollectionForm}
+							>
+								<FontAwesomeIcon icon="plus"/>
+								&nbsp;New collection
+							</Button>
+					}
+					{
+						this.state.showCollectionForm ?
+							<Button bsStyle="success" onClick={this.handleAddToNewCollection}>
+								<FontAwesomeIcon icon="plus"/> Add to new collection
+							</Button> :
+							<Button bsStyle="success" onClick={this.handleAddToCollection}>
+								<FontAwesomeIcon icon="plus"/>Add to selected collection{this.state.selectedCollections.length > 1 ? 's' : null}
+							</Button>
+					}
+					<Button bsStyle="danger" onClick={this.props.onCloseCallback}>
+						<FontAwesomeIcon icon="times"/> Cancel
```

Comment:
After some thought and using the modal, I feel like "Close" would be better than "Cancel" since if I added to a or created a collection, the button will not revert that action.
```suggestion
						<FontAwesomeIcon icon="times"/> Close
```


If I close the modal, the collection selection is not cleared (which is good IMO) but the new collection form is cleared. It's because when we toggle the collection rows it's stored in the state, while the name/description/privacy isn't.
I don't think that's an issue in this case, but something to keep in mind I guess.